hibernate-core/src/main/java/org/hibernate/id/Configurable.java

```diff
@@ -34,11 +42,13 @@ public interface Configurable {
 	 *
 	 * @param type The id property type descriptor
 	 * @param parameters param values, keyed by parameter name
-	 * @param serviceRegistry Access to service that may be needed.
+	 * @param context Access to the generator creation context.
 	 *
 	 * @throws MappingException when there's something wrong with the given parameters
 	 */
-	void configure(Type type, Properties parameters, ServiceRegistry serviceRegistry) throws MappingException;
+	default void configure(Type type, Properties parameters, GeneratorCreationContext context) throws MappingException {
```

Comment:
I believe the `Type` is available from the `GeneratorCreationContext` via `context.getProperty().getType()`, so I would not repeat it here. I would make the signature:

    void configure(GeneratorCreationContext serviceRegistry, Properties parameters)

WDYT?

Reply:
That's even better! I've updated the method signature and the various implementations. I've also introduced a default `getType()` method in `GeneratorCreationContext`, so you can still override it (mainly useful for tests).